pkg/kubectl/cmd/logs.go

```diff
@@ -224,26 +262,41 @@ func (o LogsOptions) Validate() error {
 }
 
 // RunLogs retrieves a pod log
-func (o LogsOptions) RunLogs() error {
-	switch t := o.Object.(type) {
-	case *api.PodList:
-		for _, p := range t.Items {
-			if err := o.getLogs(&p); err != nil {
-				return err
+func (o LogsOptions) RunLogs(out io.Writer, args []string) error {
+	if len(o.PrefixMatchList) > 1 {
+		firstPrint := true
+		fmt.Fprintf(out, "No pod named %s, did you mean? ", args[0])
+		for i := range o.PrefixMatchList {
```

Comment:
Still not entirely sure how this would look if a large number of pods matched the prefix given.
@fabianofranz maybe there should be a limit on the amount of pods we suggest?